src/rules/avoid-importing-barrel-files.ts

```diff
@@ -0,0 +1,332 @@
+import { readFileSync } from 'node:fs'
+import { builtinModules } from 'node:module'
+import path from 'node:path'
+
+import {
+  count_module_graph_size,
+  is_barrel_file,
+} from 'eslint-barrel-file-utils/index.cjs'
+import { ResolverFactory } from 'unrs-resolver'
+
+import { createRule } from '../utils/index.js'
+
+export interface Options {
+  allowList: string[]
+  maxModuleGraphSizeAllowed: number
+  amountOfExportsToConsiderModuleAsBarrel: number
+  debug: boolean
+  exportConditions: string[]
+  mainFields: string[]
+  extensions: string[]
+  tsconfig?: {
+    configFile: string
+    references: string[]
+  }
+  alias?: Record<string, Array<string | null | undefined>>
+}
+
+export type MessageId = 'avoidImport'
+
+const defaultOptions: Options = {
+  allowList: [],
+  maxModuleGraphSizeAllowed: 20,
+  amountOfExportsToConsiderModuleAsBarrel: 3,
+  debug: false,
+  exportConditions: ['node', 'import'],
+  mainFields: ['module', 'browser', 'main'],
+  extensions: ['.js', '.ts', '.tsx', '.jsx', '.json', '.node'],
```

Comment:
For first version without releasing it's acceptable, but it should be replaced by `import-x`'s own resolver eventually.

Reply:
it does use `unrs-resolver`

can you clarify what you mean?